dds/DCPS/XTypes/DynamicTypeSupport.cpp

```diff
@@ -10,36 +10,51 @@
 
 #  include "DynamicTypeImpl.h"
 #  include "Utils.h"
+#  include "DynamicDataImpl.h"
 
 #  include <dds/DCPS/debug.h>
 #  include <dds/DCPS/DCPS_Utils.h>
 
+using namespace OpenDDS::DCPS;
+
 OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 namespace OpenDDS {
 namespace XTypes {
 
-bool DynamicSample::serialize(DCPS::Serializer& ser) const
+bool DynamicSample::serialize(Serializer& ser) const
 {
-  ACE_UNUSED_ARG(ser);
-  // TODO
-  return false;
+  const DynamicDataImpl* const ddi = dynamic_cast<DynamicDataImpl*>(data_.in());
+  if (!ddi) {
+    if (log_level >= LogLevel::Notice) {
+      ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: DynamicSample::serialize: "
+        "currently we only support DyanmicDataImpl\n"));
```

Comment:
Beside from a typo (Dyanmic) - I suggest eventually, once we've implemented it, rewording this (along with the similar one in `serialized_size`) to mention DynamicDataFactory since that's what the user should access.

Reply:
Yeah, "dyanmic" is apparently something subconsciously I keep writing. I've had to grep for it a few times in the last few months.